libsplinter/src/service/network/interconnect/mod.rs

```diff
@@ -0,0 +1,719 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod error;
+mod scml;
+
+use std::collections::HashMap;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::thread;
+
+use crate::matrix::{MatrixReceiver, MatrixRecvError, MatrixSender};
+use crate::network::dispatch::{
+    ConnectionId, DispatchLoopBuilder, DispatchLoopShutdownSignaler, DispatchMessageSender,
+    Dispatcher, MessageSender,
+};
+use crate::protos::component::{ComponentMessage, ComponentMessageType};
+
+pub use self::error::{ServiceInterconnectError, ServiceLookupError};
+
+/// The ServiceLookup trait provides an interface for looking up details about individual service
+/// connections.
+pub trait ServiceLookup: Send {
+    /// Retrieves the connection id for a given service id, if found.
+    ///
+    /// # Errors
+    ///
+    /// Returns a ServiceLookupError if the connection id cannot be retrieved.
+    fn connection_id(&self, service_id: &str) -> Result<Option<String>, ServiceLookupError>;
+
+    /// Retrieves the service id for a given connection id, if found.
+    ///
+    /// # Errors
+    ///
+    /// Returns a ServiceLookupError if the service id cannot be retrieved.
+    fn service_id(&self, connection_id: &str) -> Result<Option<String>, ServiceLookupError>;
+}
+
+pub trait ServiceLookupProvider {
+    fn service_lookup(&self) -> Box<dyn ServiceLookup>;
+}
+
+/// ServiceInterconnect will receive incoming messages from services and dispatch them to the
+/// ComponentMessageType handlers. It will also receive messages from handlers that need to be
+/// sent to components.
+///
+/// When an incoming message is received, the connection_id is converted to a service processor
```

Comment:
Change to either "the connection ID" or connection_id. 

Pro tip: When referring to something by the (code) name, don't use "the".  When using words (not code literals), "the" or "a" should be used.